tests/e2e/lib/dpa_helpers.go

```diff
@@ -112,6 +114,13 @@ func (v *DpaCustomResource) Build(backupRestoreType BackupRestoreType) *oadpv1al
 		dpaSpec.Configuration.Velero.DefaultPlugins = append(dpaSpec.Configuration.Velero.DefaultPlugins, oadpv1alpha1.DefaultPluginCSI)
 		dpaSpec.Configuration.Velero.FeatureFlags = append(dpaSpec.Configuration.Velero.FeatureFlags, velero.CSIFeatureFlag)
 		dpaSpec.SnapshotLocations = nil
+	case VSL:
+		dpaSpec.SnapshotLocations[0].Velero.Credential = &corev1.SecretKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: v.VSLSecretName,
+			},
+			Key: "cloud",
+		}
```

Comment:
also need to remove BSLs, right?

```go
dpaSpec.BackupLocations = nil
```

Reply:
no, why? 

BSL + VSL is fine right.

Reply:
just to be sure VSL is used? (for the other tests, VSLs are removed from DPA) I do not have experience with VSLs, some other maintainers input would be good

Reply:
For other tests, VSLs are removed from the DPA. Because, the volumes are being backed up using other ways like csi, file system backup.

In the VSL case, Other resources are still backed up using BSL.
Just the volumes are being backed up using native snapshots workflow.

I am quite sure about that.

Reply:
@shubham-pampattiwar @kaovilai can you please confirm.

Reply:
BSLs are independent from VSL. To use VSL do not require removal of BSL.

Reply:
If you use CSI, or file system backup, you do not need VSL. VSL is to allow use of velero cloud provider plugin "native snapshots" which do not require csi driver or a running pod on a pvc.

Reply:
So, I am still confused. How Velero will know to use BSL or VSL to perform the backup?

Reply:
It is not BSL or VSL

BSL always need to be there, to backup other K8s resources.

It is just about the volume backup part

If we specify, defaultVolumestoFsBackup -> it will do file system backup using restic/kopia

If we specify, snapshotMoveData -> it will perform csi snapshot with native datamover

If both are false, and VSL is there in spec, it will use Native Snapshot (VSL)

Reply:
spec.VolumeSnapshotLocations is by default added, and used in case other methods of volume backups are false. So should not be needed to change this function.

Reply:
So, after discussion with the team, I was wrong about removal of BSL, it is needed for native snapshots backups

And yes, velero automatically adds bsl/vsl ref to backup spec